src/app/icd/ICDConfigurationData.cpp

```diff
@@ -36,4 +37,16 @@ System::Clock::Milliseconds32 ICDConfigurationData::GetSlowPollingInterval()
     return mSlowPollingInterval;
 }
 
+CHIP_ERROR ICDConfigurationData::SetModeDurations(uint32_t activeModeDuration_ms, uint32_t idleModeInterval_s)
+{
+    VerifyOrReturnError(activeModeDuration_ms <= (idleModeInterval_s * 1000), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Pretty sure the System::Clock types would handle this for you without having to count the 0s on your constant:
```
  Seconds32 s(1);
  Milliseconds32 ms(500);
  printf("Less than: %d\n", ms < s);
```
prints "1" for the boolean.